metrics/fidelity/utility.py

```diff
@@ -0,0 +1,174 @@
+import logging
+
+import numpy as np
+import torch
+
+
+logger = logging.getLogger(__name__)
+
+
+def compute_fidelity(prob_y_hat, prob_y_hat_alpha, fidelity_type="sufficiency", clip=True, dataset_level=False):
+    """
+    clip=False ERASER's definition of sufficiency and comprehensiveness Cite?? ask chenhao
```

Comment:
Don't leave an "ask chenhao" comment in the final version

I added all the relevant citations at the beginning. Just explain what clipping means and refer to our paper.